### PR TITLE
Hoist the multi-buffer broadcast from the transform stack

### PR DESCRIPTION
**Pull request by giuseros**
We should hoist the `Broadcast` out of the multi-buffer transform stack, in this way we won't affect the stack invertibility. THis means that the `Broadcast` transform becomes a `i%multiBufferFactor` affine map.

### CODE REVIEW

**Review by manupak — COMMENTED** [review]
Code and the idea on high-level make sense to me!
Do you have a simple unit test to demo this ?

**Comment by giuseros** [comment]
Yep, you are right. I amended the tests and added a check to make sure that the lowering to  `transforming_for`s is now producing the right (vectorized) strides

File: mlir/test/Dialect/Rock/test_multi_buffer.mlir [hunk 1]
```diff
@@ -73,7 +74,11 @@ module attributes {mhal.arch = "amdgcn-amd-amdhsa:gfx90a"} {
     %18 = arith.divui %17, %14 : index
     rock.threadwise_read_into {forceUnroll, useIndexDiffs} [](%2) [%c0, %9, %16, %18, %6] -> %7 : memref<16x1x6x16x256x16xf16> -> memref<16xf16, #gpu.address_space<private>>
     rock.threadwise_read_into {forceUnroll, useIndexDiffs} [](%4) [%c0, %9, %16, %18, %6] -> %8 : memref<16x1x6x16x256x16xf16> -> memref<16xf16, #gpu.address_space<private>>
-    %19 = rock.alloc() : memref<16xf16, #gpu.address_space<private>>
+    %c0_0 = arith.constant 0 : index
+    // LOWERING: %[[multibuf0:.*]] = rock.alloc() : memref<64xi8, #gpu.address_space<private>>
+    %raw = rock.alloc(){__multibuffer__=2} : memref<32xi8, #gpu.address_space<private>>
```
Comment (krzysz00) [inline mlir/test/Dialect/Rock/test_multi_buffer.mlir:79]: Nit: Can you make this an inherent attribute on `rock.alloc` so it's harder to forget about?

**Review by krzysz00 — APPROVED** [review]
Seems fine

File: mlir/test/Dialect/Rock/test_multi_buffer.mlir (hunk 1 above)
Comment (giuseros) [reply mlir/test/Dialect/Rock/test_multi_buffer.mlir:79]: We chatted about that, and the point is that `__multibuffer__` attribute is only used by the `--rock-multibuffer-test` pass.

**Review by manupak — APPROVED** [review]
LGTM!

**Pull request merged**